filebeat/input/filestream/internal/input-logfile/publish.go

```diff
@@ -128,7 +128,7 @@ func (op *updateOp) Execute(n uint) {
 	resource.stateMutex.Lock()
 	defer resource.stateMutex.Unlock()
 
-	if resource.lockedVersion != op.resource.version {
+	if resource.lockedVersion != op.resource.version || resource.isDeleted() {
```

Comment:
There is still `writeState` that can add some state to the stored registry. Do we need to protect the persistentStore in `writeState` as well?

Reply:
Good point. I have updated the function to invalidate the resource after it has been persisted.